File: src/main/java/org/railwaystations/rsapi/core/model/PublicInboxEntry.java
package org.railwaystations.rsapi.core.model;

import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class PublicInboxEntry {

    String countryCode;

    String stationId;

    String title;

    Coordinates coordinates;

    public Double getLat() {
        return coordinates != null ? coordinates.getLat() : null;
    }

    public Double getLon() {
        return coordinates != null ? coordinates.getLon() : null;
    }

}

File: src/main/kotlin/org/railwaystations/rsapi/core/model/PublicInboxEntry.kt
package org.railwaystations.rsapi.core.model

data class PublicInboxEntry(
    var countryCode: String,
    var stationId: String,
    var title: String,
    var coordinates: Coordinates,
) {

    val lat: Double
        get() = coordinates.lat

    val lon: Double
        get() = coordinates.lon
}
